modules/docs/mdx/7.0-MIGRATION-GUIDE.mdx

```diff
@@ -538,6 +539,242 @@ main `react` package.
 - `Flex` was promoted from `@workday/canvas-kit-labs/layout` to `@workday/canvas-kit-react/layout`
 - `Stack` was promoted from `@workday/canvas-kit-labs/layout` to `@workday/canvas-kit-react/layout`
 
+## Model Changes
+
+### Guards and Callbacks
+
+We changed the signature of model event guards and callbacks. In v6, the parameters were in an
+object. This has a less than ideal developer experience as intellisense isn't engaged immediately
+and we don't plan on adding any additional parameters to guard and callback functions. We've removed
+the object wrapper.
+
+🤖 The codemod handles these changes automatically.
+
+Before:
+
+```tsx
+const model = useTabsModel({
+  onSelect({data: {id}, prevState}) {
+    console.log(id, prevState);
+  },
+});
+```
+
+After:
+
+```tsx
+const model = useTabsModel({
+  onSelect({id}, prevState) {
+    console.log(id, prevState);
+  },
+});
+```
+
+### Model Implementation
+
+If you don't extend models, you can skip this section.
+
+In v6 we supported Typescript 3.8 which limited the way types could be inferred and defined. Canvas
+Kit v7 now requires Typescript 4.1 which introduced
+[Template Literal Types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html).
+This means that the guards and callbacks no longer need to be manually defined via an `eventMap`.
+Event maps were a stopgap and were manual and suffered from user-error in defining them. Not all
+events had guards and callbacks there's no check other than manual review to ensure the existence of
+an event callback or guard. Template Literal Types allow us to properly type guards and callbacks
+without event maps. The previous types forced `data` to be an object. We've since dropped that
+restriction. You can still only pass a single argument to events. If you need additional
```

Comment:
> You can still only pass a single argument to events. If you need additional
information, use an object.

I am actually a little confused reading this. I wonder if some of this should move past the next example so the code changes are presented before reading this. This doesn't refer to code in the previous example, does it?

Reply:
All events can only take a single parameter because of the signature of guards and callbacks:

```ts
// event signature
type updateValue = (value: string) => void

// guard signature - note the first parameter always gets named `data`
type shouldUpdateValue = (data: string, state: State) => boolean

// callback signature
type onUpdateValue = (data: string, prevState: State) => void
```

The first parameter of the event signature becomes the first parameter of a guard and callback. If we allowed more than a single argument, we'd run into issues:

```ts
// event signature
type updateValue = (value: string, delay = false) => void

// guard - only takes first parameter
type shouldUpdateValue = (data: string, state: State) => boolean

// callback signature - only takes first parameter
type onUpdateValue = (data: string, prevState: State) => void
```

The previous example would have to look like this:
```ts
// event signature
type updateValue = (event: { value: string, delay: boolean }) => void

// guard - only takes first parameter
type shouldUpdateValue = (data: { value: string, delay: boolean }, state: State }) => boolean

// callback signature - only takes first parameter
type onUpdateValue = (data: { value: string, delay: boolean }, prevState: State) => void
```